### PR TITLE
feat: Add more metrics and error log for observability publisher

### PR DESCRIPTION
**Pull request by tiopramayudi**
# Description

Add kafka lag metrics and error log for easier troubleshoot when issue arise regarding observability publisher
# Modifications

* Add scheduled daemon to calculate lag for kafka consumer 
* Add more error log during failure
* Add container port for observability publisher deployment so it can be scraped by the prometheus agent
# Tests

# Checklist
- [ ] Added PR label
- [ ] Added unit test, integration, and/or e2e tests
- [ ] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduces API changes

# Release Notes

```release-note

```

### CODE REVIEW

File: python/observation-publisher/publisher/prediction_log_consumer.py [hunk 1]
```diff
@@ -141,9 +142,54 @@ def __init__(
...
         self._batch_size = config.batch_size
+        self._topic = config.topic
         self._consumer.subscribe([config.topic])
         self._poll_timeout = config.poll_timeout_seconds
 
+        background_job_thread = Thread(target=self._emit_metrics)
+        background_job_thread.setDaemon(True)
+        background_job_thread.start()
+
+    def _emit_metrics(self):
```
Comment (bthari) [inline python/observation-publisher/publisher/prediction_log_consumer.py:153]: I want to confirm something, is the objective of this function is to calculate the kafka lag every 60 seconds?

Comment (tiopramayudi) [reply python/observation-publisher/publisher/prediction_log_consumer.py:153]: yes

Comment (deadlycoconuts) [reply python/observation-publisher/publisher/prediction_log_consumer.py:153]: > please reconfirm what is the acceptable interval

Are we going to remove this comment in https://github.com/caraml-dev/merlin/pull/598/files#diff-95854776df8c83183573e19f3bf6e2bfe61f73ccc4077d92ee9b64385c13d403R159 then? 😅

File: api/pkg/observability/deployment/deployment.go [hunk 2]
```diff
@@ -394,14 +394,21 @@ func (c *deployer) createDeploymentSpec(ctx context.Context, data *models.Worker
...
+							},
 						},
 					},
 					Volumes: []corev1.Volume{
 						{
 							Name: cfgVolName,
 							VolumeSource: corev1.VolumeSource{
 								Secret: &corev1.SecretVolumeSource{
-									SecretName: c.getSecretName(data),
+									SecretName: secretName,
```
[the remark this answers is not in the document]
Comment (tiopramayudi) [reply api/pkg/observability/deployment/deployment.go:411]: no, it just to use everything that is passed to the function, the caller use `getSecretName` as `secretName` parameter in this function so the value would be the same.

[the remark this answers is not in the document]
Comment (deadlycoconuts) [reply api/pkg/observability/deployment/deployment.go:411]: Ah got it, just noticed it was previously unused. Thanks!

File: python/observation-publisher/publisher/prediction_log_consumer.py (hunk 1 above)
Comment (bthari) [reply python/observation-publisher/publisher/prediction_log_consumer.py:153]: when will this function be called repeatedly to log the lag, afaik the `start()` on `Thread` only call the target function once?

Comment (tiopramayudi) [reply python/observation-publisher/publisher/prediction_log_consumer.py:153]: nice catch! it should be looping forever in target function

**Review by deadlycoconuts — APPROVED** [review]
Thanks a lot for the changes @tiopramayudi and @bthari for reviewing this! Don't think I have any other comments on the PR 🚀

**Comment by bthari** [comment]
Thank you for the changes! LGTM 🚀

**Pull request merged**